test/test_cli.rb

```diff
@@ -1,441 +1,357 @@
+            assert_equal 100, Sidekiq.options[:concurrency]
+            assert_equal '/tmp/sidekiq.pid', Sidekiq.options[:pidfile]
+            assert_equal '/tmp/sidekiq.log', Sidekiq.options[:logfile]
+            assert_equal 7, Sidekiq.options[:queues].count { |q| q == 'often' }
+            assert_equal 3, Sidekiq.options[:queues].count { |q| q == 'seldom' }
+          end
+        end
       end
     end
 
-    describe 'Sidekiq::CLI#parse_queues' do
-      describe 'when weight is present' do
-        it 'concatenates queues by factor of weight and sets strict to false' do
-          opts = { strict: true }
-          @cli.__send__ :parse_queues, opts, [['often', 7], ['repeatedly', 3]]
-          @cli.__send__ :parse_queues, opts, [['once']]
-          assert_equal (%w[often] * 7 + %w[repeatedly] * 3 + %w[once]), opts[:queues]
-          assert !opts[:strict]
+    describe 'validation' do
+      describe 'when required application path does not exist' do
+        it 'exits with status 1' do
+          exit = assert_raises(SystemExit) { @cli.parse(%w[sidekiq -r /non/existent/path]) }
+          assert_equal 1, exit.status
         end
       end
 
-      describe 'when weight is not present' do
-        it 'returns queues and sets strict' do
-          opts = { strict: true }
-          @cli.__send__ :parse_queues, opts, [['once'], ['one_time']]
-          @cli.__send__ :parse_queues, opts, [['einmal']]
-          assert_equal %w[once one_time einmal], opts[:queues]
-          assert opts[:strict]
+      describe 'when required path is a directory without config/application.rb' do
+        it 'exits with status 1' do
+          exit = assert_raises(SystemExit) { @cli.parse(%w[sidekiq -r ./test/fixtures]) }
+          assert_equal 1, exit.status
         end
       end
     end
 
-    describe 'Sidekiq::CLI#parse_queue' do
-      describe 'when weight is present' do
-        it 'concatenates queue to opts[:queues] weight number of times and sets strict to false' do
-          opts = { strict: true }
-          @cli.__send__ :parse_queue, opts, 'often', 7
-          assert_equal %w[often] * 7, opts[:queues]
-          assert !opts[:strict]
-        end
-      end
+    it 'does not require the specified Ruby code' do
+      @cli.parse(%w[sidekiq -r ./test/fake_env.rb])
 
-      describe 'when weight is not present' do
-        it 'concatenates queue to opts[:queues] once and leaves strict true' do
-          opts = { strict: true }
-          @cli.__send__ :parse_queue, opts, 'once', nil
-          assert_equal %w[once], opts[:queues]
-          assert opts[:strict]
-        end
-      end
+      refute($LOADED_FEATURES.any? { |x| x =~ /fake_env/ })
+    end
+
+    it 'does not boot rails' do
+      refute defined?(::Rails::Application)
+
+      @cli.parse(%w[sidekiq -r ./myapp])
+
+      refute defined?(::Rails::Application)
     end
   end
 
-  describe 'misc' do
+  describe '#run' do
```

Comment:
I've left a couple of skipped (TODO) tests to not forget about them.

Reply:
were you going to supply these tests or are these TODO for someone else?

Reply:
I'm definitely going to cover `#run` with tests ^_^ checkout the code precisely at the moment and think how to do it better. I guess it's possible to test some parts for non-daemonized mode with corresponding refactoring and mocking. Do you have the suggestions?

Reply:
No suggestions.  If you aren't going to implement something yourself, please remove the TODOs.

Reply:
TBH, I don't see any drawbacks to specify some parts of the lib require the testing. But I don't mins to remove them in this PR and will add in the next one related to `#run` tests.

Reply:
Done